### PR TITLE
Feature/dapp allow to disable ethereum providers

### LINKED ISSUE #2742: Only show Direct RPC Provider if it has been enabled
## Description

Since using a Direct RPC Provider is not recommended for security reasons we should make a user explicitly enable it if the want to use it.

## Acceptance criteria
- Button should not be available by default. The button should only be there ones a user has manually enabled that option.

## Tasks
- [ ] Enable via the configuration file, should not be enabled by default
- [ ] The button should not be visible if this functionality is disabled

### PR DESCRIPTION
**Pull request by weilbith**
**Thank you for submitting this pull request :)**

Fixes #2742

**Short description**
The last commit includes some code refactorings of the ConnectionManager component. Its got got too cluttered now and it was time to take action. It was a known issue.

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. Run the dApp in development mode and verify all three providers are presented
2. Run the dApp in production mode and verify that only two providers are presented (private key is missing)

### CODE REVIEW

File: raiden-dapp/src/components/ConnectionManager.vue [hunk 1]
```diff
@@ -137,6 +145,18 @@ export default class ConnectionManager extends Vue {
     return DirectRpcProvider.isAvailable;
   }
 
+  async created(): Promise<void> {
+    this.checkProvidersForBeingDisabled();
+  }
+
+  async checkProvidersForBeingDisabled(): Promise<void> {
```
Comment (taleldayekh) [inline raiden-dapp/src/components/ConnectionManager.vue]: `checkDisabledProviders`

**Review by taleldayekh — APPROVED** [review]
Works like a charm, great job 👍🏽

File: raiden-dapp/src/components/ConnectionManager.vue (hunk 1 above)
Comment (weilbith) [reply raiden-dapp/src/components/ConnectionManager.vue]: I would like to insist on my naming. Yours is more like "checking the disabled providers". But it is checking all providers to see which of them are disabled.

Comment (taleldayekh) [reply raiden-dapp/src/components/ConnectionManager.vue]: Sure, as I already approved, feel free to go with this naming  but fyi it is grammatically wrong :) `checkForDisbaledProviders` would be more correct too say the same thing.

Comment (weilbith) [reply raiden-dapp/src/components/ConnectionManager.vue]: Perfect. Thanks! 🙏🏾

Comment (weilbith) [reply raiden-dapp/src/components/ConnectionManager.vue]: ✔️

**Comment by weilbith** [comment]
Due to the refactoring of the ConnectionManager component the dApp's end-to-end tests fail. Actually they class for the buttons are still there. Let's see what the actual issue is.

**Comment by weilbith** [comment]
I'll now abuse this PR a little and run the e2e job many times to see if it has become more stable now.

**Pull request merged**